src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/EnvConfigurationsNav.tsx

```diff
@@ -0,0 +1,281 @@
+                Icon: ICAdd,
+                props: {
+                    className: 'fcn-6',
+                },
+                btnProps: {
+                    onClick: onHeaderIconBtnClick(EnvResourceType.ConfigMap),
+                },
+            },
+            items: updatedEnvConfig.configMaps,
+            noItemsText: 'No configmaps',
+            isExpanded: expandedIds?.configmap,
+        },
+        {
+            header: 'Secrets',
+            id: EnvResourceType.Secret,
+            headerIconConfig: {
+                Icon: ICAdd,
+                props: {
+                    className: 'fcn-6',
+                },
+                btnProps: {
+                    onClick: onHeaderIconBtnClick(EnvResourceType.Secret),
+                },
+            },
+            items: updatedEnvConfig.secrets,
+            noItemsText: 'No secrets',
+            isExpanded: expandedIds?.secrets,
+        },
+    ]
+
+    // REACT SELECT PROPS
+    const envOptions = [
+        {
+            environmentName: 'Base Configurations',
+            environmentId: -1,
+            isProtected: isBaseConfigProtected,
+        },
+        ...(isDevtronApp ? environments : jobEnvs),
+    ]
+
+    const onEnvSelect = ({ environmentId }: typeof environmentData) => {
+        if (isDevtronApp) {
+            history.push(getNavigationPath(path, appId, environmentId, EnvResourceType.DeploymentTemplate))
+        } else {
+            history.push(
+                getNavigationPath(
+                    path,
+                    appId,
+                    environmentId,
+                    EnvResourceType.ConfigMap,
+                    undefined,
+                    updatedEnvConfig.configMaps[0]?.title,
+                ),
+            )
+        }
+    }
+
+    const renderEnvSelector = () => {
+        return (
+            <div className="flexbox dc__align-center dc__gap-8 p-12 dc__border-bottom en-1">
+                <NavLink to={lastUnlockedStage}>
+                    <div className="dc__border br-4 flex p-1">
+                        <ICBack className="icon-dim-16" />
+                    </div>
+                </NavLink>
+                <ReactSelect<typeof environmentData>
+                    classNamePrefix="env-config-selector"
+                    isSearchable={false}
+                    isClearable={false}
+                    value={environmentData}
+                    options={envOptions}
+                    getOptionLabel={(option) => `${option.environmentName}`}
+                    getOptionValue={(option) => `${option.environmentId}`}
+                    styles={envSelectStyles}
+                    components={{
+                        IndicatorSeparator: null,
+                        Option,
+                        DropdownIndicator: EnvSelectDropdownIndicator,
+                    }}
+                    onChange={onEnvSelect}
+                    placeholder="Select Environment"
```

Comment:
Do we need isLoading here along with isDisabled? If yes, please use the LoadingIndicator component as well in the components object

Reply:
not required